R/sampleSummaryStats.R

```diff
@@ -1,134 +1,140 @@
-#' @export
-setGeneric("sampleSummaryStatsTable", function(inSCE, ...) standardGeneric("sampleSummaryStatsTable"))
-
-#' @export
-setGeneric("sampleSummaryStatsTable<-", function(inSCE, ..., value) standardGeneric("sampleSummaryStatsTable<-"))
-
-#' @export
+#' @rdname sampleSummaryStatsTable
 setMethod("sampleSummaryStatsTable", "SingleCellExperiment", function(inSCE, statsName, ...){
-    return(inSCE@metadata$sctk$sampleSummary[[statsName]])
+    allStatsNames <- listSampleSummaryStatsTables(inSCE)
+    if(length(allStatsNames) == 0){
+        stop(paste("No tables are stored within the SingleCellExperiment object."),
+             "Please execute the sampleSummaryStats function first.")
+    }
+    if(!statsName %in% allStatsNames){
+        stop(paste(statsName,
+                   "is not a table within the SingleCellExperiment object.",
+                   "The following are the names of the tables stored:",
+                   allStatsNames))
```

Comment:
Would be better to have paste(allStatsNames, collapse=",") at the end of the stop string.